spec.bs

```diff
@@ -4117,6 +4149,7 @@ navigating to another page. Some implementations, such as Chromium, have chosen
     Note: Implementations can consider loading only a portion of these interest groups
     at a time to avoid issuing too many requests at once.
     1. Let |ig| be a deep copy of |originalInterestGroup|.
+    1. Run [=update k-anonymity cache for interest group=] for |ig|.

```

Comment:
So from looking into https://github.com/WICG/turtledove/pull/1005#discussion_r1597122418 more, I think the answer is no? This line is not necessarily running in parallel right? The update interest groups algorithm can run on the main thread [I think](https://wicg.github.io/turtledove/#ref-for-interest-group-update), which means my suggestion cannot be made. So I think we should undo the parallel assertions. I like adding them when possible, but it takes some digging to determine whether they are valid or not. But usually I find that up-front digging is worth it for spec readability and spotting logic bugs as the spec evolves.

Reply:
Hmm.. It probably should be in parallel. Nothing waits for it to happen and it makes network requests that could block.

Reply:
Maybe it should be, but I'm saying today it cannot be. Assuming my comment above is correct. That is, if the update interest group algorithm can indeed run on the main thread, and can invoke this algorithm (also on the main thread), then the in parallel assertion is broken.

Reply:
In the link you referenced, the update is occurring in the context of the AbortSignal which is not in the main thread, so I think it should still be in parallel.

Reply:
Abort algorithms attached to AbortSignals always run on the main thread though.

Reply:
Ah, I didn't know that. It's not supposed to be on the main thread, so that's a bug in the spec. For now, I'll just remove the assertions for "in parallel" then.

Reply:
From the abort steps, you could easily enter an in parallel context to kick off the subroutines it currently runs on the main thread.

Reply:
Done